x-pack/plugins/security/server/authentication/providers/anonymous.ts

```diff
@@ -105,10 +105,8 @@ export class AnonymousAuthenticationProvider extends BaseAuthenticationProvider
     return new HTTPAuthorizationHeader(
       'Basic',
       new BasicHTTPAuthorizationHeaderCredentials(
-        // @ts-expect-error upgrade typescript v5.1.6
-        credentials.username,
-        // @ts-expect-error upgrade typescript v5.1.6
-        credentials.password
+        (credentials as UsernameAndPasswordCredentials).username,
```

Comment:
__issue:__ we already have a type-guard function above that was supposed to narrow type down to `UsernameAndPasswordCredentials` - `isAPIKeyCredentials`, but TS 5.1 got stricter and we need to improve our type guard (add `ReadOnly<...>` in two places) instead of force-casting here:
```ts
function isAPIKeyCredentials(
  credentials: Readonly<
    ElasticsearchAnonymousUserCredentials | APIKeyCredentials | UsernameAndPasswordCredentials
  >
): credentials is Readonly<APIKeyCredentials> {
  ...
}
```

Reply:
😅 gahh that makes sense. Idk what I was thinking!